src/olympia/activity/models.py

```diff
@@ -238,19 +233,13 @@ def for_groups(self, groups):
         return self.filter(grouplog__group__in=groups)
 
     def for_user(self, user):
-        vals = (UserLog.objects.filter(user=user)
-                .values_list('activity_log', flat=True))
-        return self.filter(pk__in=list(vals))
+        return self.filter(userlog__user=user)
 
     def for_block(self, block):
-        vals = (BlockLog.objects.filter(block=block)
-                .values_list('activity_log', flat=True))
-        return self.filter(pk__in=list(vals))
+        return self.filter(blocklog__block=block)
 
     def for_guidblock(self, guid):
-        vals = (BlockLog.objects.filter(guid=guid)
-                .values_list('activity_log', flat=True))
-        return self.filter(pk__in=list(vals))
+        return self.filter(blocklog__block__guid=guid)
```

Comment:
Why is the one above `blocklog__block=block` but this is `blocklog__block__guid=guid`? Is `blocklog__guid` not a thing? I ask because the old versions use `BlockLog.objects.filter(block=block)` and `BlockLog.objects.filter(guid=guid)`.

Reply:
Good catch, I went a little too quickly on this one. It shouldn't change functionality but does save us a join.